src/ring_theory/ideal/local_ring.lean

```diff
@@ -371,6 +371,53 @@ lemma map_comp (f : T →+* R) (g : R →+* S) [is_local_ring_hom f] [is_local_r
   (local_ring.residue_field.map g).comp (local_ring.residue_field.map f) :=
 ideal.quotient.ring_hom_ext $ ring_hom.ext $ λx, rfl
 
+section
+
+open ring_equiv local_ring
+
+@[simp] lemma map_id_apply {x : residue_field R} : map (ring_hom.id R) x = x :=
+by simp only [map_id, ring_hom.id_apply]
+
+@[simp] lemma map_map (f : R →+* S) (g : S →+* T) (x : residue_field R)
+[is_local_ring_hom f] [is_local_ring_hom g] :
+  map g (map f x) = map (g.comp f) x :=
+by simp only [map_comp, ring_hom.comp_apply]
+
+noncomputable theory
+
+/-- A ring isomorphism defines an isomorphism of residue fields -/
+@[simps apply]
+def map_equiv (f : R ≃+* S):
+  local_ring.residue_field R ≃+* local_ring.residue_field S :=
+{ to_fun := map (f : R →+* S),
+  inv_fun := map (f.symm : S →+* R),
+  left_inv := λ x, by simp only [map_map, symm_comp, map_id, ring_hom.id_apply],
+  right_inv := λ x, by simp only [map_map, comp_symm, map_id, ring_hom.id_apply],
+  map_mul' := ring_hom.map_mul _,
+  map_add' := ring_hom.map_add _ }
+
+@[simp]
+lemma map_equiv.symm (f : R ≃+* S):
+  (map_equiv f).symm = map_equiv f.symm := ext $ λ x, rfl
+
+/-- The group homomorphism from `ring_aut R` to `ring_aut k` where `k`
+is the residue field of `R`. -/
+def map_aut:
+  ring_aut R →* ring_aut (local_ring.residue_field R):=
+{ to_fun := map_equiv,
+  map_mul' := begin
+  intros φ ψ,
+  ext,
```

Comment:
Does `exact map_comp (φ : R →+* R) (ψ : R →+* R)` work here?

Reply:
No, I doesn't seem to work

Reply:
I meant here _instead_ of the `ext` line

Reply:
I' ve pushed the code with the error - let me know what I could try

Reply:
Oh, OK. Still getting the same error- do I need a new lemma for that?

Reply:
Oh, you still need `ext : 1` here; only the second step of the `ext` is unhelpful

Reply:
Ah you're right, it's easiest to keep the `ext`.


Have a look at the original error message; again, the lemma you're applying is in the opposite direction to your goal.

Reply:
OK, thanks. it compiles now